subprojects/core/src/main/java/org/gradle/api/internal/tasks/options/OptionValueNotationParserFactory.java

```diff
@@ -16,24 +16,31 @@
 
 package org.gradle.api.internal.tasks.options;
 
+import org.gradle.api.file.FileSystemLocation;
 import org.gradle.internal.Cast;
 import org.gradle.internal.exceptions.DiagnosticsVisitor;
+import org.gradle.internal.typeconversion.DoubleFromCharSequenceNotationConverter;
 import org.gradle.internal.typeconversion.EnumFromCharSequenceNotationParser;
 import org.gradle.internal.typeconversion.IntegerFromCharSequenceNotationConverter;
+import org.gradle.internal.typeconversion.LongFromCharSequenceNotationConverter;
 import org.gradle.internal.typeconversion.NotationConverter;
 import org.gradle.internal.typeconversion.NotationConverterToNotationParserAdapter;
 import org.gradle.internal.typeconversion.NotationParser;
 
 public class OptionValueNotationParserFactory {
     public <T> NotationParser<CharSequence, T> toComposite(Class<T> targetType) throws OptionValidationException {
-        if (targetType.isAssignableFrom(String.class)) {
+        if (targetType.isAssignableFrom(String.class) || targetType.isAssignableFrom(FileSystemLocation.class)) {
             return Cast.uncheckedCast(new NoDescriptionValuesJustReturningParser());
```

Comment:
❓ I don't understand why we are using the same parser here for `String` and `FileSystemLocation`. How is this possible?